beam/www/beam/pages/Repack.vue

```diff
@@ -8,24 +8,293 @@
 }
+
+const loadBOMs = async () => {
+	const boms = await store.getAll<{ name: string }[]>('BOM')
+	bomList.value = boms.map(bom => bom.name)
+}
+
+const loadWarehouses = async () => {
+	const warehouses = await store.getAll<{ name: string }[]>('Warehouse', {
+		filters: JSON.stringify([['is_group', '!=', '1']]),
+	})
+	warehouseList.value = warehouses.map(warehouse => warehouse.name)
+}
+
+const clearField = (field: 'from_warehouse' | 'to_warehouse') =>
+	store.$patch(state => (state.cache.mappers.repack[field] = ''))
+
+const addCurrentItem = () => currentItem.value.qty++
+const substractCurrentItem = () => (currentItem.value.qty > 0 ? currentItem.value.qty-- : 0)
+const clearCurrentItem = (field: 'item_code' | 'bom') => (currentItem.value[field] = '')
+
+const create = async () => {
+	const body: StockEntry = {
+		stock_entry_type: 'Repack',
+		items: items.value,
+		name: stockEntry.value.name,
+	}
+	let res: DocActionResponse<StockEntry>
+
+	if (body.name) {
+		res = await store.update<StockEntry>('Stock Entry', body.name, body)
+	} else {
+		res = await store.insert<StockEntry>('Stock Entry', body)
+	}
+	const { data, response } = res
+	if (data?.name) store.$patch(state => (state.cache.mappers.repack.name = data.name))
+
+	return { data, response }
+}
+
+const repack = async () => {
+	const res = await store.submit<StockEntry>('Stock Entry', stockEntry.value.name || '')
+	if (res?.data) {
+		store.$patch(state => {
+			state.cache.mappers.repack = {
+				name: '',
+				stock_entry_type: 'Repack',
+				items: [],
+				from_warehouse: '',
+				to_warehouse: '',
+			}
+		})
+		items.value = []
+	}
+}
+
+const addItem = () => {
+	if (!currentItem.value.item_code || !currentItem.value.qty) {
+		toast.error('Please select or scan an Item and set its quantity')
+		return
+	}
+	if (!stockEntry.value.from_warehouse && !stockEntry.value.to_warehouse) {
+		toast.error('Please select source or target warehouses')
+		return
+	}
+	if (stockEntry.value.from_warehouse && stockEntry.value.to_warehouse) {
+		toast.error('Please select only source or target warehouse')
+		return
+	}
+	items.value.push({
+		label: currentItem.value.item_code,
+		count: { count: currentItem.value.qty },
+		description: stockEntry.value.from_warehouse
+			? `From ${stockEntry.value.from_warehouse}`
+			: `To ${stockEntry.value.to_warehouse}`,
+		item_code: currentItem.value.item_code,
+		qty: currentItem.value.qty,
+		s_warehouse: stockEntry.value.from_warehouse,
+		t_warehouse: stockEntry.value.to_warehouse,
+	})
+
+	currentItem.value = { item_code: '', qty: 0, bom: '' }
+	clearField('from_warehouse')
+	clearField('to_warehouse')
+}
+
+const clearItem = () => {
+	currentItem.value.bom = ''
+	items.value = []
+	store.$patch(state => (state.cache.mappers.repack.items = []))
+}
+
+const controlButtons = computed((): ControlButton[] => {
+	const buttons = [
+		{ label: 'CLEAN', color: { background: '#4791FF', text: 'var(--sc-btn-color)' }, action: clearItem },
```

Comment:
I don't think this button should show up until an item has been added, otherwise it might get confusing